src/graphics.rs

```diff
@@ -470,18 +473,19 @@ impl GlCache {
             self.stored_vertex_buffer = self.vertex_buffer;
         } else {
             self.stored_index_buffer = self.index_buffer;
+            self.stored_index_type = self.index_type;
         }
     }
 
     fn restore_buffer_binding(&mut self, target: GLenum) {
         if target == GL_ARRAY_BUFFER {
-            if self.stored_vertex_buffer != 0 {
-                self.bind_buffer(target, self.stored_vertex_buffer);
+            if self.stored_vertex_buffer != 0 && self.stored_vertex_buffer != target {
```

Comment:
Hmm, I did not quite get that new check. What is its purpose?

Reply:
When you update the buffer with new IndexType it will store current index_type `None`, update it and then restore with stored `None`.